tests/simpleKVBC/TesterReplica/internalCommandsHandler.cpp

```diff
@@ -88,17 +94,35 @@ void InternalCommandsHandler::execute(InternalCommandsHandler::ExecutionRequests
                                    req.outActualReplySize,
                                    req.outReplicaSpecificInfoSize);
     } else {
-      res = executeWriteCommand(req.requestSize,
-                                req.request,
-                                req.executionSequenceNum,
-                                req.flags,
-                                req.maxReplySize,
-                                req.outReply,
-                                req.outActualReplySize);
+      if (block_accumulation_enabled && (!pre_execute && (req.flags & bftEngine::MsgFlag::HAS_PRE_PROCESSED_FLAG))) {
+        res = executeBlockAccumulatedWriteCommand(req.requestSize,
```

Comment:
It can be  made to a single function but with checks for block accumulation while  updating keys, replies and adding blocks as all these are done differently or later.

Reply:
Another option is to take the common code parts to specific functions. My point is to avoid a code duplication.

Reply:
Sure. Will make changes

Reply:
Updated code with changes removing the new method and added some refactored methods to avid code duplication.